include/ideep/operators/conv.hpp

```diff
@@ -1496,7 +1509,98 @@ struct convolution_forward
           aalgorithm, aprop_kind, x_dtype, src_dims, attr, aengine);
     }
   }
+#ifdef __aarch64__
+  template <bool with_bias>
+  static std::pair<primitive_desc, dnnl::convolution_forward> get_primitive_desc(
+      const tensor::desc& src_desc,
+      const tensor::desc& weights_desc,
+      const tensor::desc& bias_desc,
+      const tensor::desc& dst_desc,
+      const dims& strides,
+      const dims& dilates,
+      const dims& padding_l,
+      const dims& padding_r,
+      const size_t weights_hashkey = 0, /*this is to check inplace weight updates*/
+      bool is_channels_last = false,
+      const attr_t& attr = attr_t(),
+      algorithm aalgorithm = algorithm::convolution_direct,
+      prop_kind aprop_kind = prop_kind::forward,
+      const engine& aengine = engine::cpu_engine()) {
+    auto src_desc_query = src_desc;
+    auto weights_desc_query = weights_desc;
+    auto bias_desc_query = with_bias ? bias_desc : tensor::desc();
+    auto dst_desc_query = dst_desc;
+
+    src_desc_query = src_desc.to_format_any();
+    weights_desc_query = weights_desc.to_format_any();
+    bias_desc_query = with_bias ? bias_desc.to_format_any() : tensor::desc();
+    dst_desc_query = dst_desc.to_format_any();
+
+    // For nhwc path, weight uses format_tag::any,
+    // while activation uses format_tag::nhwc.
+    auto ndims = src_desc.get_dims().size();
+    if (is_channels_last) {
+      auto memory_format = tag::nhwc;
+      if (3 == ndims) {
+        memory_format = tag::nwc;
+      } else if (5 == ndims) {
+        memory_format = tag::ndhwc;
+      }
+      src_desc_query = src_desc.to_format(memory_format);
+      dst_desc_query = dst_desc.to_format(memory_format);
+    }
+
+    auto key = utils::create_key(
+        aprop_kind,
+        aalgorithm,
+        src_desc_query,
+        weights_desc_query,
+        bias_desc_query,
+        dst_desc_query,
+        with_bias,
+        strides,
+        dilates,
+        padding_l,
+        padding_r,
+        attr,
+        omp_get_max_threads(),
+        weights_hashkey);
+
+    dnnl::convolution_forward::primitive_desc pd;
+    if (with_bias) {
+      pd = primitive_desc(
+            {aprop_kind,
+             aalgorithm,
+             src_desc_query,
+             weights_desc_query,
+             bias_desc_query,
+             dst_desc_query,
+             strides,
+             dilates,
+             padding_l,
+             padding_r},
+            attr,
+            aengine);
+    } else {
+      pd = primitive_desc(
+            {aprop_kind,
+             aalgorithm,
+             src_desc_query,
+             weights_desc_query,
+             dst_desc_query,
+             strides,
+             dilates,
+             padding_l,
+             padding_r},
+            attr,
+            aengine);
+      }
 
+    return fetch_or_create(key, [&]() {
+      return std::make_pair(pd, super(pd));
```

Comment:
You may consider to put the pd creation (i.e., `primitive_desc()`) here to avoid additional overhead, up to you.